app/models/user.rb

```diff
@@ -445,6 +446,24 @@ def self.try_to_auto_create_user(login, password)
     user
   end
 
+  def password_changed
+    @password_changed ||= nil
+  end
+
+  def password_changed=(value)
+    attribute_will_change!('password_changed') if password_changed != value
+    @password_changed = value
+  end
+
+  def password_changed_changed?
+    changed.include?('password_changed')
+  end
+
+  def set_password_changed
+    self.password_changed = true
```

Comment:
This seems to be missing a conditional, will it not mark password_changed on _every_ model validation?